src/Native/System.Native/pal_mount.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "../config.h"
+#include "pal_mount.h"
+#include "pal_utilities.h"
+#include <assert.h>
+#include <string.h>
+#include <errno.h>
+
+// Check if we should use getmntinfo or /proc/mounts
+#if HAVE_MNTINFO
+#include <sys/mount.h>
+#else
+#include <sys/statfs.h>
+#include <mntent.h>
+#define STRING_BUFFER_SIZE 8192
+#endif
+
+#if HAVE_MNTINFO
+
+int GetMountInfo(MountPointFound onFound)
+{
+    // getmntinfo returns pointers to OS-internal structs, so we don't need to worry about free'ing the object
+    struct statfs* mounts = nullptr;
+    int count = getmntinfo(&mounts, 0);
+    for (int32_t i = 0; i < count; i++)
+    {
+        onFound(mounts[i].f_mntonname);
+    }
+
+    return 0;
+}
+
+#else
+
+int GetMountInfo(MountPointFound onFound)
+{
+    int result = 0;
+    FILE* fp = setmntent("/proc/mounts", MNTOPT_RO);
+    if (fp == nullptr)
+    {
+        result = errno;
+    }
+    else
+    {
+        // The _r version of getmntent needs all buffers to be passed in; however, we don't know how big of a string
+        // buffer we will need, so pick something that seems like it will be big enough.
+        char buffer[STRING_BUFFER_SIZE] = { };
+        mntent entry = { };
```

Comment:
The zero'ing of the entry isn't, however I'd like to keep the buffer zero'ing in there - it always worries me alloc'ing a buffer I'm going to read from without zero'ing it, even if it should be overwritten. If you think that's excessive I can remove it